app/controllers/patients_controller.rb

```diff
@@ -242,53 +242,170 @@ def update
     render json: patient
   end
 
-  def update_hoh
+  # Moves a record into a household
+  def move_to_household
     new_hoh_id = params.permit(:new_hoh_id)[:new_hoh_id]&.to_i
-    current_patient_id = params.permit(:id)[:id]
-    household_ids = params[:household_ids] || []
+    current_patient_id = params.permit(:id)[:id]&.to_i
+
     current_patient = current_user.get_patient(current_patient_id)
-    old_hoh = current_user.get_patient(current_patient.responder_id)
     new_hoh = current_user.get_patient(new_hoh_id)
+    current_user_patients = current_user.patients
+
+    # ----- Error Checking -----
+
+    # Check to make sure selected HoH record exists.
+    unless current_user_patients.exists?(new_hoh_id)
+      error_message = "Move to household action failed: selected Head of Household with ID #{new_hoh_id} is not accessible."
+      render(json: { error: error_message }, status: :forbidden) && return
+    end
 
-    # Don't do anything if there hasn't been a change to the responder at all
+    # Check to make sure user has access to update this record.
+    unless current_user_patients.exists?(current_patient_id)
+      error_message = 'Move to household action failed: user does not have permissions to update current monitoree.'
+      render(json: { error: error_message }, status: :forbidden) && return
+    end
+
+    # Do not do anything if there hasn't been a change to the responder at all.
     redirect_to(root_url) && return if current_patient.responder_id == new_hoh_id
```

Comment:
should we make it so that the list of suggestions for household members does not include the current patient? not sure how often this might happen in production, but when I ran into this case, I was confused for a bit before I realized I was trying to set HoH as the current patient since no error message shows up and nothing happens other than the page being refreshed

![Screen Shot 2021-01-28 at 4 54 46 PM](https://user-images.githubusercontent.com/9956561/106203543-8b468300-6189-11eb-9714-8953897c7ec8.png)

Reply:
Agreed that would be ideal. I may be able to just do a check to exclude this record after the `patients_table_data` method returns in the `head_of_household_options` action. I'll work on that.

Reply:
@timwongj I've made a separate ticket for this - it needs some more changes I don't want to shove into this PR.
https://tracker.codev.mitre.org/browse/SARAALERT-1273